composer/algorithms/alibi/attention_surgery_functions/_bert.py

```diff
@@ -0,0 +1,157 @@
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+import math
+from typing import Optional, Tuple
+from types import MethodType
+
+import torch
+from torch import nn
+
+from composer.algorithms.alibi.attention_surgery_functions.utils import (
+    register_surgery_function_builder,
+    register_alibi,
+    zero_and_freeze_expand_position_embeddings
+)
+from transformers.models.bert.modeling_bert import BertSelfAttention, BertEmbeddings
+from transformers.models.roberta.modeling_roberta import RobertaSelfAttention, RobertaEmbeddings
+
+@register_surgery_function_builder(BertEmbeddings, RobertaEmbeddings)
+def build_bert_embedding_converter(max_sequence_length: int):
+    """Builds a function to remove positional embeddings.
+    
+    The built function also expands `position_ids` buffer to support `max_sequence_length` tokens.
+    """
+    def convert_position_embeddings_and_ids(module: torch.nn.Module, module_index: Optional[int] = None):
```

Comment:
Should `module`'s type specifically be `Union[BertEmbeddings, RobertaEmbeddings]`? Or can we rely on that based on how the registry is implemented?

Reply:
We can rely on that based on how `alibi.py` will use the registry, but if someone wanted to hack into the registry to get a hold of these functions then they could pass arbitrary `torch.nn.Module` instances. I think this hypothetical person understands the risk of their misadventure, but, for the sake of good measure, I'll set the more restrictive type (which will also elucidate the way the code works).